### PR TITLE
Add comments and improve code quality

### PR DESCRIPTION
**Pull request by joaosimbiose**

### CODE REVIEW

File: src/main/java/com/slicingdice/jslicer/SlicingDice.java [hunk 1]
```diff
@@ -48,6 +48,10 @@
  */
 public class SlicingDice {
 
+    private static final String POST = "post";
+    private static final String PUT = "put";
+    private static final String DELETE = "delete";
+
     private String _apiKey;
     private String _masterKey;
     private String _customKey;
```
Comment (rafael-telles) [inline src/main/java/com/slicingdice/jslicer/SlicingDice.java]: why these underscores? remove it

File: src/main/java/com/slicingdice/jslicer/SlicingDice.java [hunk 2]
```diff
@@ -100,33 +106,34 @@ public Headers getHeaders() {
         return this.headers;
     }
 
-    private ArrayList<Object> getCurrentKey(){
+    private ArrayList<Object> getCurrentKey() throws SlicingDiceKeyException {
         if (this._masterKey != null){
-            ArrayList<Object> result = new ArrayList<Object>();
```
Comment (rafael-telles) [inline src/main/java/com/slicingdice/jslicer/SlicingDice.java]: refactor this method x.x

File: src/main/java/com/slicingdice/jslicer/SlicingDice.java [hunk 3]
```diff
@@ -203,72 +215,131 @@ private boolean checkRequest(Response response) throws SDHttpError {
...
         return null;
     }
-    public JSONObject createField(JSONObject data) throws IOException {
-        String url = BaseURL + URLResources.FIELD.url;
+
+    /**
+     * Create field in Slicing Dice
+     *
+     * @param data A JSONObject in the Slicing Dice field format
+     * @return A String with json request result if your field is valid
```
Comment (rafael-telles) [inline src/main/java/com/slicingdice/jslicer/SlicingDice.java]: it doesn't return a string...

File: src/main/java/com/slicingdice/jslicer/SlicingDice.java [hunk 4]
```diff
@@ -203,72 +215,131 @@ private boolean checkRequest(Response response) throws SDHttpError {
...
+     *
+     * @param data A JSONObject in the Slicing Dice field format
+     * @param test if true the field will be created on test end-point otherwise on production
+     *             end-point
+     * @return A String with json request result if your field is valid
+     * @throws IOException
+     */
+    public JSONObject createField(final JSONObject data, final boolean test) throws IOException {
         String url = this.wrapperTest(test);
         url += URLResources.FIELD.url;
```
Comment (rafael-telles) [inline src/main/java/com/slicingdice/jslicer/SlicingDice.java]: final String url = this.wrapperTest(test) + URLResources.FIELD.url;

File: src/main/java/com/slicingdice/jslicer/SlicingDice.java [hunk 5]
```diff
@@ -203,72 +215,131 @@ private boolean checkRequest(Response response) throws SDHttpError {
...
+    /**
+     * Get all fields.
+     *
+     * @param test if true the request will be to test end-point
+     * @return All fields(active and inactive).
+     * @throws IOException
+     */
+    public JSONObject getFields(final boolean test) throws IOException {
         String url = this.wrapperTest(test);
         url += URLResources.FIELD.url;
```
Comment (rafael-telles) [inline src/main/java/com/slicingdice/jslicer/SlicingDice.java]: final String url = this.wrapperTest(test) +  URLResources.FIELD.url;

File: src/main/java/com/slicingdice/jslicer/SlicingDice.java [hunk 6]
```diff
@@ -203,72 +215,131 @@ private boolean checkRequest(Response response) throws SDHttpError {
...
 
-    public JSONObject index(JSONObject data) throws IOException {
-        String url = BaseURL + URLResources.INDEX.url;
-        return this.makeRequest(url, data, "post", 1);
+    /**
+     * Index data to existing entities or create new entities, if necessary. This method corresponds
+     * to a POST request at /index.
+     *
+     * @param data A JSON object in the SlicingDice index format
+     * @return A String with json request result if your indexation is valid
```
Comment (rafael-telles) [inline src/main/java/com/slicingdice/jslicer/SlicingDice.java]: doesn't return string

**Review by rafael-telles — CHANGES_REQUESTED** [review]

File: src/main/java/com/slicingdice/jslicer/utils/validators/FieldValidator.java [hunk 7]
```diff
@@ -48,55 +48,62 @@ public FieldValidator(JSONObject data) {
...
+    private void validateFieldDescription() throws InvalidFieldDescriptionException {
+        final String description = this.data.getString("description");
         if (description.trim().length() == 0) {
-            throw new InvalidFieldDescriptionException("The field's description can't be empty/None.");
+            throw new InvalidFieldDescriptionException(
+                    "The field's description can't be empty/None.");
         } else if (description.length() > 300) {
-            throw new InvalidFieldDescriptionException("The field's description have a very big name.(Max: 300 chars)");
+            throw new InvalidFieldDescriptionException(
+                    "The field's description have a very big name.(Max: 300 chars)");
```
Comment (telles-simbiose) [inline src/main/java/com/slicingdice/jslicer/utils/validators/FieldValidator.java]: weird sentence, the description has a name?

**Review by telles-simbiose — APPROVED** [review]

**Pull request merged**